lib/mongodb.js

```diff
@@ -1963,7 +1969,7 @@ function isObjectIDProperty(modelCtor, propDef, value, options) {
   if (typeof value === 'string' && value.match(ObjectIdValueRegex)) {
     if (isStoredAsObjectID(propDef)) return true;
     else return !isStrictObjectIDCoercionEnabled(modelCtor, options);
-  } else if (value instanceof ObjectID) {
+  } else if (value instanceof mongodb.ObjectID) {
```

Comment:
`ObjectID` is a factory function that creates `new mongodb.ObjectID`, it's not a class. Therefore the expression `value instanceof ObjectID` is never true.